service/src/main/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderService.java

```diff
@@ -145,11 +147,21 @@ public OrderTypeAndDocument buildOrderTypeAndDocument(OrderTypeAndDocument typeA
                         caseData.getCaseLocalAuthority(), orderTypeAndDocument.hasInterimSubtype()));
                 break;
             case SUPERVISION_ORDER:
-                orderTemplateBuilder
-                    .put("orderTitle", orderTypeAndDocument.getFullType())
-                    .put("childrenAct", "Section 31 and Paragraphs 1 and 2 Schedule 3 Children Act 1989")
-                    .put("orderDetails", getFormattedFinalSupervisionOrderDetails(getChildrenDetails(caseData).size(),
-                        caseData.getCaseLocalAuthority(), caseData.getOrderMonths()));
+                if (subtype == INTERIM) {
+                    orderTemplateBuilder
+                        .put("orderTitle", orderTypeAndDocument.getFullType(INTERIM))
+                        .put("childrenAct", "Section 38 and Paragraphs 1 and 2 Schedule 3 Children Act 1989")
+                        .put("orderDetails",
+                            getFormattedInterimSupervisionOrderDetails(getChildrenDetails(caseData).size(),
+                                caseData.getCaseLocalAuthority()));
+                } else if (subtype == FINAL) {
```

Comment:
Minor but could just be else, it is either going to be `INTERIM` or `FINAL` and nothing else